To: vim_dev@googlegroups.com
Subject: Patch 7.3.305
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.305
Problem:    Auto-loading a function while editing the command line causes
	    scrolling up the display.
Solution:   Don't set msg_scroll when defining a function and the user is not
	    typing. (Yasuhiro Matsumoto)
Files:	    src/eval.c


*** ../vim-7.3.304/src/eval.c	2011-08-17 15:23:16.000000000 +0200
--- src/eval.c	2011-09-14 14:28:21.000000000 +0200
***************
*** 20786,20792 ****
      nesting = 0;
      for (;;)
      {
! 	msg_scroll = TRUE;
  	need_wait_return = FALSE;
  	sourcing_lnum_off = sourcing_lnum;
  
--- 20786,20793 ----
      nesting = 0;
      for (;;)
      {
! 	if (KeyTyped)
! 	    msg_scroll = TRUE;
  	need_wait_return = FALSE;
  	sourcing_lnum_off = sourcing_lnum;
  
*** ../vim-7.3.304/src/version.c	2011-09-14 10:49:44.000000000 +0200
--- src/version.c	2011-09-14 14:30:49.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     305,
  /**/

-- 
If someone questions your market projections, simply point out that your
target market is "People who are nuts" and "People who will buy any damn
thing".  Nobody is going to tell you there aren't enough of those people
to go around.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
